experimental/c-cpp/runner.py

```diff
@@ -0,0 +1,302 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+RUN apt-get update && apt-get install -y make autoconf automake libtool cmake \
+                      pkg-config curl check
+RUN rm /usr/local/bin/cargo && \
+ curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y && \
+ apt-get install -y cargo
+RUN python3 -m pip install --upgrade pip && \
+    python3 -m pip install pydantic-core pyyaml cxxfilt openai==1.16.2
+COPY *.py $SRC/
+WORKDIR $SRC
+COPY build.sh $SRC/"""
+
+empty_project_yaml = """homepage: "https://github.com/google/oss-fuzz"
+language: c++
+primary_contact: "info@oss-fuzz.com"
+auto_ccs:
+-
+main_repo: 'https://github.com/samtools/htslib.git'
+"""
+
+
+def setup_worker_project(oss_fuzz_base: str, project_name: str):
+  """Setup empty OSS-Fuzz project used for analysis."""
+  temp_project_dir = os.path.join(oss_fuzz_base, "projects", project_name)
+  if os.path.isdir(temp_project_dir):
+    shutil.rmtree(temp_project_dir)
+
+  os.makedirs(temp_project_dir)
+  with open(os.path.join(temp_project_dir, "project.yaml"), 'w') as f:
+    f.write(empty_project_yaml)
+  with open(os.path.join(temp_project_dir, "build.sh"), 'w') as f:
+    f.write(empty_oss_fuzz_build)
+  with open(os.path.join(temp_project_dir, "Dockerfile"), 'w') as f:
+    f.write(empty_oss_fuzz_docker)
+
+  # Copy over the generator
+  shutil.copyfile(
+      os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                   "build-generator.py"),
+      os.path.join(temp_project_dir, "build-generator.py"))
+
+  # Build a version of the project
+  if silent_global:
+    subprocess.check_call("python3 infra/helper.py build_fuzzers %s" %
+                          (project_name),
+                          shell=True,
+                          cwd=oss_fuzz_base,
+                          stdout=subprocess.DEVNULL,
+                          stderr=subprocess.DEVNULL)
+  else:
+    subprocess.check_call("python3 infra/helper.py build_fuzzers %s" %
+                          (project_name),
+                          shell=True,
+                          cwd=oss_fuzz_base)
+
+
+def run_autogen(github_url,
+                outdir,
+                openai_api_key,
+                oss_fuzz_base,
+                worker_project,
+                disable_autofuzz,
+                targets_per_heuristic=5):
+  """Launch auto-gen analysis within OSS-Fuzz container."""
+
+  initiator_cmd = 'python3 /src/build-generator.py %s -o %s' % (github_url,
+                                                                outdir)
+  if disable_autofuzz:
+    initiator_cmd += ' --disable-fuzzgen'
+  initiator_cmd += ' --targets-per-heuristic=%d' % (targets_per_heuristic)
+
+  cmd = [
+      "docker",
```

Comment:
nit: can we use single quotes for all strings other than docstrings?

Reply:
Yup, done.